t/18begin_work_without_raise_error.t

```diff
@@ -0,0 +1,47 @@
+use strict;
+use warnings;
+
+use Test::More;
+use DBI;
+
+use vars qw($test_dsn $test_user $test_password);
+use lib 't', '.';
+require 'lib.pl';
+
+my $dbh = DbiTestConnect($test_dsn, $test_user, $test_password, { RaiseError => 0, PrintError => 1 });
+
+plan tests => 12;
+
+my $warn;
+$SIG{__WARN__} = sub { $warn = $_[0]; note "Caught DBI warn: $warn"; };
+
+my $error;
+$SIG{__DIE__} = sub { $error = $_[0]; note "Caught DBI error: $error"; };
+
+# shutdown needs Perl file handle but mariadb_sockfd() returns C file descritor number
+# mode "+<&" assign a dup() copy of C file descriptor to Perl file handle and
+# mode "+<&=" assign directly C file descriptor to Perl file handle
+# close automatically close C file descriptor in Perl file handle
+# mysql client library does not expect if somebody closes its file descriptors
+# so always take a copy of mariadb_sockfd() C file descriptor and just shutdown it
+open my $socket, '+<&', $dbh->mariadb_sockfd();
+ok(shutdown($socket, 2), "shutdown socket handle");
+close $socket;
```

Comment:
This is a copy'n'paste of the other commit in 15reconnect.t. Do we need the same test twice?

Reply:
In 18begin_work_without_raise_error.t is checked that begin_work works correctly after network failure when RaiseError is turned off. And this shutdown() call here simulate network failure. I added comment into this test.

In 15reconnect.t is checked that reconnect by `mariadb_auto_reconnect` works correctly also after simulated network failure.

So 15reconnect.t and 18begin_work_without_raise_error.t tests do different things. First expects success and second expects failure.

Reply:
Any chance the same part of the code plus the comment went into a lib file?

Reply:
Ok, moved it into t/lib.pl